pkg/providers/tinkerbell/tinkerbell.go

```diff
@@ -75,21 +81,32 @@ type ProviderTinkClient interface {
 	GetHardware(ctx context.Context) ([]*tink.Hardware, error)
 }
 
-func NewProvider(datacenterConfig *v1alpha1.TinkerbellDatacenterConfig, machineConfigs map[string]*v1alpha1.TinkerbellMachineConfig, clusterConfig *v1alpha1.Cluster, providerKubectlClient ProviderKubectlClient, providerTinkClient ProviderTinkClient, now types.NowFunc, skipIpCheck bool, hardwareConfigFile string) *tinkerbellProvider {
+type ProviderPbnjClient interface {
+	ValidateBMCSecretCreds(ctx context.Context, bmc pbnj.BmcSecretConfig) error
+}
+
+func NewTinkerbellClient(ProviderTinkClient ProviderTinkClient, ProviderPbnjClient ProviderPbnjClient) *TinkerbellClients {
+	return &TinkerbellClients{
+		ProviderTinkClient, ProviderPbnjClient,
+	}
+}
+
+func NewProvider(datacenterConfig *v1alpha1.TinkerbellDatacenterConfig, machineConfigs map[string]*v1alpha1.TinkerbellMachineConfig, clusterConfig *v1alpha1.Cluster, providerKubectlClient ProviderKubectlClient, providerTinkbellClient TinkerbellClients, now types.NowFunc, skipIpCheck bool, hardwareConfigFile string) *tinkerbellProvider {
 	return NewProviderCustomNet(
 		datacenterConfig,
 		machineConfigs,
 		clusterConfig,
 		providerKubectlClient,
-		providerTinkClient,
+		providerTinkbellClient.ProviderTinkClient,
+		providerTinkbellClient.ProviderPbnjClient,
 		&networkutils.DefaultNetClient{},
 		now,
 		skipIpCheck,
 		hardwareConfigFile,
 	)
 }
 
-func NewProviderCustomNet(datacenterConfig *v1alpha1.TinkerbellDatacenterConfig, machineConfigs map[string]*v1alpha1.TinkerbellMachineConfig, clusterConfig *v1alpha1.Cluster, providerKubectlClient ProviderKubectlClient, providerTinkClient ProviderTinkClient, netClient networkutils.NetClient, now types.NowFunc, skipIpCheck bool, hardwareConfigFile string) *tinkerbellProvider {
+func NewProviderCustomNet(datacenterConfig *v1alpha1.TinkerbellDatacenterConfig, machineConfigs map[string]*v1alpha1.TinkerbellMachineConfig, clusterConfig *v1alpha1.Cluster, providerKubectlClient ProviderKubectlClient, providerTinkClient ProviderTinkClient, pbnjClient ProviderPbnjClient, netClient networkutils.NetClient, now types.NowFunc, skipIpCheck bool, hardwareConfigFile string) *tinkerbellProvider {
```

Comment:
nit: Can we rename this `NewProviderCustomNet`? I think it was orginally named this because the only additional dependency injected was `networkutils.NetClient`. 
We now inject a few more.

Reply:
This function has too many clients being injected. It suggests the construct it returns, given this is an initializing constructor func, has too much responsibility. Have we thought about how we can break the provider down?

Reply:
Yah I agree. We could possibly break out all the external clients and have them injected somewhere else and the constructor would only target it’s members. 
But yes again another discussion 👍🏻